cfpb/sbl-frontend
# Filing process routing - Phase 2 (Live data)

Closes #404  

## Changes
- [Filing] Automatically creates a new `Filing` when needed
- [Filing overview] Determines latest incomplete Filing step and links to it
- [Filing sub-pages] Creates placeholder pages for Errors, Warnings, Point of contact, and Complete
- [Filing sub-pages] Displays the status of all Steps at the top of each sub-page
- [Styling] Updates step label's font-weight and color based on the steps status (complete, incomplete)
- [Styling] Only displays a step as Complete (Dark blue bar, semibold font) when a user has navigated beyond the step in the Filing flow (i.e. When viewing `Warnings`, `Errors` would show complete)

## How to test this PR

1. Start fresh
2. [Landing page](http://localhost:8899/filing) - Verify all associated institutions are ready for Upload (which indicates that the Filing for this period has automatically been created)
3. Upload a ["failing"](https://github.com/cfpb/regtech-data-validator/tree/main/tests/data) data file.  Verify that you *cannot* navigate beyond the `Upload` page (`Save..` button disabled)
4. Upload a ["passing"](https://github.com/cfpb/regtech-data-validator/tree/main/tests/data) data file. Verify that you *can* navigate to the `Errors` page  (`Save..` button enabled). Also verify that the step indicator at the top of the page updates to reflect the status of upcoming steps. 
5. Return to the [Landing page](http://localhost:8899/filing), verify that the Institution that you have uploaded data for now links to the latest "Incomplete" step (probably "Point of Contact").

## Screenshots (Updated 4/25)
https://github.com/cfpb/sbl-frontend/assets/2592907/54280e03-a2e5-4a10-b0d2-5a4127bac342
